libnetdata/threads/threads.c

```diff
@@ -109,6 +109,33 @@ static void thread_cleanup(void *ptr) {
     netdata_thread = NULL;
 }
 
+static void thread_set_name(NETDATA_THREAD *nt) {
+
+    if (nt->tag) {
+        char *threadname = strdupz(nt->tag);
```

Comment:
The name should point directly to the string instead of copying it, i.e.
```
         char *threadname = nt->tag; 
```
The tag itself has already been `strdupz`ed in `netdata_thread_create()` and in any case this function calls `free(threadname);` which would corrupt `threadname` if it was necessary to duplicate it.

So you should also remove:
```
             free(threadname); 
```

Reply:
Can you explain your rationale here because you resolved this conversation but I still don't understand why. What I'm asking for is to remove the `strdupz()` and `free()` calls from `thread_set_name()`.

Reply:
My gh page didnt' get updated when I woke it from sleep, sorry.
Because of the limit of 15chars for `pthread_setname_np` I have to force a null line 120: `threadname[15] = 0;`, else the setname call fails with `ERANGE`

As it's a reference to a netdata_thread, I don't want to change the tag value (just for a limitation from the pthread).
Hence, I have to copy it, split it to 15chars, and free it.

Reply:
Ok I see. In that case I would prefer we not mess with heap memory any more than necessary (too much `malloc()` fragmentation), and you would also need to use `freez()` instead of `free()` for consistency reasons.

Instead I propose you use the stack and one of netdata's internal functions that take care of the trailing `'\0'`, that would be [strncpyz](https://github.com/netdata/netdata/blob/0fca36fd596e49395141561bc42c168a240f51b8/libnetdata/inlined.h#L238).

How about doing this:
```
char threadname[16];
strncpyz(threadname, tag, 15);
```
And also remove  `free()` and `threadname[15] = 0`?

Reply:
Much better indeed !